core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -629,12 +630,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       AdminUtils.createTopic(zkUtils, topic, numPartitions, replicationFactor, properties, RackAwareMode.Safe)
       info("Auto creation of topic %s with %d partitions and replication factor %d is successful"
         .format(topic, numPartitions, replicationFactor))
-      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, java.util.Collections.emptyList())
+      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, common.Topic.isInternal(topic),
```

Comment:
Not sure about that. The request classes shouldn't really have any logic IMO. They should just be a way to serialize/deserialize to the Kafka protocol. We could have a util method somewhere that does this. Or at least a static factory method in the request classes as a compromise.